coldfront/plugins/qumulo/tests/utils/test_eib_billing.py

```diff
@@ -0,0 +1,497 @@
+            """)
+            rows = cursor.fetchall()
+
+        for row in rows:
+            print(row)
+            # Confirm the initial usage is 0
+            self.assertEqual(float(row[3]) - 0, 0)
+
+        ingest_quotas_with_daily_usage()
+
+        # Exam the updated usage of the allocation
+        with connection.cursor() as cursor:
+            cursor.execute("""
+                SELECT aat.*, aau.value, aau.modified
+                FROM allocation_allocationattributeusage aau
+                JOIN allocation_allocationattribute aat
+                    ON aat.id = aau.allocation_attribute_id;
+            """) 
+            rows = cursor.fetchall()
+
+        # Exam the billing usage of the allocation from the history table
+        with connection.cursor() as cursor:
+            cursor.execute("""
+                SELECT aa.allocation_id id,
+                    aa.created,
+                    storage_filesystem_path path,
+                    aa.value quota,
+                    aau.value now_usage,
+                    ROUND(CAST(aau.value AS NUMERIC) /1024/1024/1024/1024, 2) now_usage_tb,
+                    ROUND(CAST(most_recent_haau.value AS NUMERIC) /1024/1024/1024/1024, 2) history_usage_tb,
+                    most_recent_haau.recent_modified history_timestamp
+                FROM allocation_allocationattributeusage aau
+                JOIN allocation_allocationattribute aa
+                    ON aa.id = aau.allocation_attribute_id
+                JOIN (
+                    SELECT allocation_attribute_id,
+                        MAX(modified) recent_modified,
+                        value
+                    FROM allocation_historicalallocationattributeusage
+                    GROUP BY allocation_attribute_id, DATE(modified)
+                    ) AS most_recent_haau
+                    ON aau.allocation_attribute_id = most_recent_haau.allocation_attribute_id
+                JOIN allocation_allocationattributetype aat
+                    ON aat.id = aa.allocation_attribute_type_id
+                JOIN (
+                    SELECT aa.allocation_id, aa.value storage_filesystem_path
+                        FROM allocation_allocationattribute aa
+                        JOIN allocation_allocationattributetype aat
+                           ON aa.allocation_attribute_type_id = aat.id
+                        WHERE aat.name = 'storage_filesystem_path'
+                    ) AS storage_filesystem_path
+                    ON aa.allocation_id = storage_filesystem_path.allocation_id
+                ORDER BY id DESC, history_timestamp DESC;
+            """) 
+            rows = cursor.fetchall()
+
+        for row in rows:
+            print(row)
+
+        # Confirm the status of the allocation is Active
+        for allocation in Allocation.objects.all():
+            if (allocation.resources.first().name == "Storage2"):
+                self.assertEqual(allocation.status.name, "Active")
```

Comment:
Can avoid this if you have saved the allocation from before;

```
allocation.refresh_from_db()
self.assertEqual(allocation.status.name, "Active")
```

But, a better way to do this if you needed would be:
```
allocations = Allocation.objects.filter(
    resources__name="Storage2", status__name__NE="Active"
)
self.assertEmpty(len(list(allocations)))
```

Reply:
Thanks for the suggestion. `status__name__NE` and `assertEmpty` are not defined. Instead, ```
allocations = Allocation.objects.filter(resources__name="Storage2").exclude(
            status__name__in=[
                "Active",
            ]
        )
        self.assertEqual(len(list(allocations)), 0)
```